microsoft/testsuites/storage/performance.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from enum import Enum
+from typing import Any, Dict, List
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+    schema,
+    search_space,
+    simple_requirement,
+)
+from lisa.environment import Environment
+from lisa.features import Disk
+from lisa.notifier import FIOResult, FIOResultsMessage
+from lisa.tools import Fdisk, Fio, Lscpu, Mdadm
+
+FIOMODES = Enum(
+    "FIOMODES",
+    [
+        "randread",
+        "randwrite",
+        "read",
+        "write",
+    ],
+)
+
+
+def _format_disk(
+    node: Node,
+    disk_list: List[str],
+) -> List[str]:
+    fdisk = node.tools[Fdisk]
+    partition_disks: List[str] = []
+    for data_disk in disk_list:
+        fdisk.delete_partitions(data_disk)
+        partition_disks.append(fdisk.make_partition(data_disk, format=False))
+    return partition_disks
+
+
+def _stop_raid(node: Node) -> None:
+    mdadm = node.tools[Mdadm]
+    mdadm.stop_raid()
+
+
+def _make_raid(node: Node, disk_list: List[str]) -> None:
+    mdadm = node.tools[Mdadm]
+    mdadm.create_raid(disk_list)
+
+
+@TestSuiteMetadata(
+    area="storage",
+    category="performance",
+    description="""
+    This test suite is to validate premium SSD data disks performance of Linux VM using
+     fio tool.
+    """,
+)
+class storageperf(TestSuite):  # noqa
+    TIME_OUT = 3000
```

Comment:
How long the test case will take usually?

Reply:
between 2000 and 3000